File: testing/next-dummy/app/mojo-agent.js
/**
 * Local Mojo Guardian Agent Fallback
 */
class MojoGuardian {
  constructor(apiKey) {
    this.apiKey = apiKey;
    this.endpoint = 'https://agenttestx-production-19d6.up.railway.app/api/agent/manifest';
    this.rules = {};
  }

  async init() {
    try {
      const response = await fetch(this.endpoint, {
        headers: { 'Authorization': `Bearer ${this.apiKey}` }
      });
      const data = await response.json();
      this.rules = data.rules || {};
    } catch (e) {
      console.error('Mojo Init Failed:', e);
    }
  }

  getMetadata(path) {
    return this.rules[path] || null;
  }
}

export default MojoGuardian;

File: testing/python-dummy/mojo_agent.py
import requests
import json

class MojoGuardian:
    def __init__(self, api_key):
        self.api_key = api_key
        self.endpoint = "https://agenttestx-production-19d6.up.railway.app/api/agent/manifest"
        self.rules = {}

    def init(self):
        try:
            headers = {"Authorization": f"Bearer {self.api_key}"}
            response = requests.get(self.endpoint, headers=headers, timeout=10)
            if response.status_code == 200:
                self.rules = response.json().get("rules", {})
                return True
        except Exception as e:
            print(f"[Mojo] Error: {e}")
        return False

    def get_metadata(self, path):
        clean_path = "/" if path == "" else path.rstrip("/")
        return self.rules.get(clean_path)
